File: src/main/java/com/quid/kopring/domain/book/BookRepository.java
package com.quid.kopring.domain.book;

import com.quid.kopring.book.Book;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BookRepository extends JpaRepository<Book, Long> {

    Optional<Book> findByName(String bookName);

}

File: src/main/kotlin/com/quid/kopring/book/repository/BookRepository.kt
package com.quid.kopring.book.repository

import com.quid.kopring.book.Book
import com.quid.kopring.book.model.response.BookStat

interface BookRepository {
    fun findByName(name: String): Book?
    fun getBookStat(): List<BookStat>
    fun findById(id: Long): Book?
    fun findAll(): List<Book>
    fun save(it: Book)
}